syntax = "proto3";
package iroha.ordering.proto;

import "transaction.proto";
import "proposal.proto";
import "endpoint.proto";
import "google/protobuf/empty.proto";

service OrderingGateTransportGrpc {
  rpc onProposal (protocol.Proposal) returns (google.protobuf.Empty);
}

service OrderingServiceTransportGrpc {
  rpc onTransaction (iroha.protocol.Transaction) returns (google.protobuf.Empty);
  rpc onBatch (iroha.protocol.TxList) returns (google.protobuf.Empty);
}

message TransactionsCollection {
  repeated protocol.Transaction transactions = 1;
}

message ProposalRequest {
  uint64 block_round = 1;
  uint32 reject_round = 2;
}

message ProposalResponse {
  oneof optional_proposal {
    protocol.Proposal proposal = 1;
 }
}

service OnDemandOrdering {
  rpc SendTransactions(TransactionsCollection) returns (google.protobuf.Empty);
  rpc RequestProposal(ProposalRequest) returns (ProposalResponse);
}
